compiler/noirc_evaluator/src/ssa.rs

```diff
@@ -243,6 +243,8 @@ pub fn primary_passes(options: &SsaEvaluatorOptions) -> Vec<SsaPass> {
             // A function can be potentially unreachable post-DIE if all calls to that function were removed,
             // or after the removal of unreachable instructions.
             .and_then(Ssa::remove_unreachable_functions),
+        // Another DIE pass is needed to recompute `used_globals`.
+        SsaPass::new(Ssa::dead_instruction_elimination, "Dead Instruction Elimination"),
```

Comment:
I am aware of the other PR, but couldn't we just swap the order of the removal of unreachable instruction and dead instructions?